arch/arm64/entry.S

```diff
@@ -0,0 +1,48 @@
+.section .init
+
+.type _start,@function
+.global _start
+_start:
+
+    /* According to the Linux specification */
+    /* x0 contains a pointer to the fdt */
+
+    /* Set the CPU pointer to NULL */
+    msr TPIDR_EL1, xzr
+
+    /* Set up the stack */
+    ldr x4, =__boot_stack_top
+    mov sp, x4
+
+    /* Enable floating point access */
+    mov x4, 0x330000
+    msr CPACR_EL1, x4
+    isb
+
+    /* Clear .bss */
+    ldr x4, =__bss_start
+    ldr x5, =__bss_start
+
+__bss_clear_loop:
+    strb wzr, [x5]
```

Comment:
why not store a full word at a time, its faster

Reply:
Main reason is if we are trying to boot on a small system and stop aligning sections to a page to save space. I doubt that will ever happen but this would be a nasty bug if it did. I'll do word size writes and leave a note that it assumes minimum 8 byte alignment for the section after `.bss`